File: lemma.py
# -*- coding: utf-8 -*-
import json
import sys

configFile = './rt_config.json'
config = None

with open(configFile, 'r') as config:
    config = json.load(config)

prefix = config['dictionaryDirectory']
lemmaFile = f"{config['typingsDirectory']}/{config['typingsFile']}"

lemma = input("Enter your lemma (The key you are going to use to access the translations): ")

remove = False
if len(sys.argv) > 1 and sys.argv[1] == "-r":
    remove = True

# Gather Input and initialize data
inputs = {}
data = {}

for lang in config['languages']:
    if not remove:
        inputs[lang['label']] = input(f"Enter your {lang['label']} translation: ")
    with open(f"{prefix}/{lang['label']}.json", 'r', encoding='utf8') as lang_data:
        data[lang['label']] = json.load(lang_data)


# Update data
for lang in config['languages']:
    if remove:
        data[lang['label']].pop(lemma, None)
    else:
        data[lang['label']][lemma] = inputs[lang['label']]

# Rewrite Data
for lang in config['languages']:
    with open(f"{prefix}/{lang['label']}.json", 'w+', encoding='utf8') as lang_fd:
        json.dump(data[lang['label']], lang_fd, ensure_ascii=False)


with open(lemmaFile, 'w+') as lemma:
    lemmas = ""
    i = 0
    lang_data = list(data.values())[0]
    length = len(lang_data.keys())

    for l in lang_data.keys():
        lemmas += f'"{l}"'
        if i != length - 1:
            lemmas += " | "
        i += 1

    lemma.write(f"export type Lemma = {lemmas};")

File: src/lemma.js
import * as fs from 'fs';
import { askQuestion, formatConfig } from "./util.js";

export async function lemma(lemma, remove = false) {
    const config = formatConfig();

    const prefix = config.dictionaryDirectory;
    const lemmaFile = `${config.typingsDirectory}/${config.typingsFile}`;

    // Gather Input and initialize data
    let inputs = {};
    let data = {};

    for (const lang of config.languages) {
        if (!remove)
            inputs[lang.label] = await askQuestion(`Enter your ${lang.label} translation: `);

        data[lang.label] = JSON.parse(fs.readFileSync(`${prefix}/${lang.label}.json`,
                                        { encoding: 'utf-8' }));
    }

    // Update data
    for (const lang of config.languages) {
        if (remove)
            delete data[lang.label][lemma];
        else
            data[lang.label][lemma] = inputs[lang.label];
    }

    // Rewrite data\
    
    for (const lang of config.languages) {

        fs.writeFileSync(`${prefix}/${lang.label}.json`,
            JSON.stringify(data[lang.label]),
            { encoding: 'utf-8' });
    }


    let lemmas = "";
    let i = 0;
    let lang_data = Object.values(data)[0];
    let length = Object.keys(lang_data).length;

    for (const l of Object.keys(lang_data)) {
        lemmas += `"${l}"`;

        if (i !== length - 1)
            lemmas += " | ";
        i++;
    }

    fs.writeFileSync(lemmaFile, `export type Lemma = ${lemmas};`);
}
